lib/blockchain.go

```diff
@@ -121,9 +122,19 @@ func (nn *BlockNode) IsValidateFailed() bool {
 }
 
 // IsCommitted returns true if a BlockNode has passed all validations, and it has been committed to
-// the Blockchain according to the Fast HotStuff commit rule.
+// the Blockchain according to the Fast HotStuff commit rule. This function previously considered
+// as PoW blocks to be committed regardless of status, but since we no longer keep the entire best
+// chain in-memory, we needed to update all PoW blocks in the best chain to be Committed such that
+// we could distinguish them from forks that occurred in PoW and still determine if a node is in the
+// best chain or not.
 func (nn *BlockNode) IsCommitted() bool {
-	return nn.Status&StatusBlockCommitted != 0 || !blockNodeProofOfStakeCutoverMigrationTriggered(nn.Height)
```

Comment:
Ok I see. It's if a node has been running for a long time and had a reorg at some point. Then you have no way of knowing which PoW block was actually the best one. Got it.

Reply:
yep you figured it out. Sorry I thought I dropped a comment about this somewhere.

Reply:
oh yeah the comment above this function explains it:
```
// This function previously considered
// as PoW blocks to be committed regardless of status, but since we no longer keep the entire best
// chain in-memory, we needed to update all PoW blocks in the best chain to be Committed such that
// we could distinguish them from forks that occurred in PoW and still determine if a node is in the
// best chain or not.
```